### DISCUSSION TITLE
after server transfer i am facing issue in authenticate it's always redirect me same url

### DISCUSSION THREAD

**Discussion by ghanshyam1994** [Kyon147/laravel-shopify#327] [Q&A]
https://community.shopify.com/c/shopify-apps/after-server-transfer-i-am-facing-issue-in-authenticate-it-s/m-p/2656394#M80679

please check this 

midleware "verify.shopify"

$tokenSource = $this->getAccessTokenFromRequest($request);

if ($tokenSource === null) {
//Check if there is a store record in the database
return $this->checkPreviousInstallation($request)
// Shop exists, token not available, we need to get one
? $this->handleMissingToken($request)
// Shop does not exist
: $this->handleInvalidShop($request);
}

this is alwayes redirect me to authenticate/token?....

<img src="https://community.shopify.com/c/image/serverpage/image-id/475824iAE694D67DEA2B45F/image-size/medium?v=v2&px=400" />

**Comment by shahzaibfarooq** [comment]
I also faced the same issue when I make it true "SHOPIFY_APPBRIDGE_ENABLED=true"
the issue is $target takes empty URL... To fix this is issue go to 

vendor/kyon147/laravel-shopify/src/resources/views/auth/token.blade.php

in this file replace 
{!! $target !!}{!! Str::contains($target, '?') ? '&' : '?' !!} this code to your URL  https://yourwebsite.com/shopify-apps/app-name?
